src/main/java/seedu/apollo/command/task/AddCommand.java

```diff
@@ -221,8 +224,97 @@ public boolean isClashingEvent(TaskList taskList, LocalDateTime from, LocalDateT
+
+        int startIndex = determineDay(startDay);
+        int endIndex = determineDay(endDay);
+
+        int index = startIndex;
+        do {
+            String currentDayString = currentDay.format(DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+            if (isClashingEventModule(calendar.get(index), eventStart, eventEnd, currentDayString)) {
+                ui.printClashingEventModuleMessage();
+                return;
+            }
+            index = (index + 1) % 7;
+            currentDay = currentDay.plusDays(1);
+        } while (index != endIndex);
+    }
+
+    /**
+     * Checks if an event user wants to add clashes with existing lessons.
+     *
+     * @param calendarModule The ArrayList containing the lessons.
+     * @param eventStart     The time that event starts.
+     * @param eventEnd       The time that event ends.
+     * @param currentDate    The date of which being checked.
+     * @return true if there is a clash, false otherwise.
+     */
+    private boolean isClashingEventModule(ArrayList<CalendarModule> calendarModule, LocalDateTime eventStart,
+                                          LocalDateTime eventEnd, String currentDate) {
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+        for (CalendarModule module : calendarModule) {
+            if (module.getSchedule() == null) {
+                continue;
+            }
+            String moduleStartString = currentDate + " " + module.getSchedule().getStartTime();
+            String moduleEndString = currentDate + " " + module.getSchedule().getEndTime();
+            LocalDateTime lessonStart = LocalDateTime.parse(moduleStartString, formatter);
+            LocalDateTime lessonEnd = LocalDateTime.parse(moduleEndString, formatter);
+
+            if (isOverlap(eventStart, eventEnd, lessonStart, lessonEnd)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if an event and lesson overlap.
+     *
+     * @param eventStart The start time of the event.
+     * @param eventEnd   The end time of the event.
+     * @param lessonStart The start time of the lesson.
+     * @param lessonEnd   The end time of the lesson.
+     * @return true if there is a clash, false otherwise.
+     */
+    private boolean isOverlap(LocalDateTime eventStart, LocalDateTime eventEnd, LocalDateTime lessonStart,
+                              LocalDateTime lessonEnd) {
 
+        if (eventStart.isEqual(lessonStart) && eventEnd.isEqual(lessonEnd)) {
+            return true;
+        }
+
+        if (eventStart.isBefore(lessonStart) && eventEnd.isAfter(lessonEnd)) {
+            return true;
+        }
+
+        if (eventStart.isAfter(lessonStart) && eventEnd.isBefore(lessonEnd)) {
+            return true;
+        }
+
+        if (eventStart.isBefore(lessonStart) && eventEnd.isBefore(lessonEnd)) {
+            return true;
+        }
```

Comment:
This can be simplified like the previous clashing ones, just check if the eventEnd.isBefore(lessonStart) or eventStart.isAfter(lessonEnd), return false